R/axis.R

```diff
@@ -76,9 +77,25 @@ axis.gsplot <- function(object, ..., n.minor=0, tcl.minor=0.15, reverse=NULL) {
   
   for(side in sides){
     # append the side and give it defaults if it doesn't exist
+    
     object <- modify_side(object, args = list(), side=side) 
+    which.axis <- which(names(object[[as.side_name(side)]])== 'axis')
+    if (append){
+      last.axis <- tail(which.axis, 1) # get the last one
+      object[[as.side_name(side)]] <- append(object[[as.side_name(side)]], 
+                                             list('axis' = set_args('axis', side=side, package='graphics')), 
+                                             after = last.axis)
+      object[[as.side_name(side)]][[last.axis+1]] <- append_replace(object[[as.side_name(side)]][[last.axis+1]], user_args[[fun.name]])
+    } else {
+      # remove
+      if (length(which.axis) > 1){
+        # remove all axis functions other than the first one
+        object[[as.side_name(side)]] <- object[[as.side_name(side)]][-which.axis[!which.axis %in% head(which.axis, 1)]]  
```

Comment:
My reasoning was this:
using the `append` arg to get multiple axis calls in there means appending them to the original call, which is the default set-up when the side is created. When you then set `append = FALSE` (what it is checking for here), those appended axis calls are wiped out. 

I can see your point too, where you would want to treat it like append was FALSE the whole time and use the last, but that isn't the same behavior either.

For example:
```r
gs <- gsplot() %>% 
    points(0:1,0:1) %>% 
    axis(side=1, at=c(0.5,1), labels=FALSE) %>% 
    axis(side=1, at=c(0.25, 0.75), line = 2, append=TRUE) %>% 
    axis(side=1, at=c(0.45, 0.55), lwd = 2, append=TRUE) %>% 
    axis(side=1, at=c(0.33))
```
the last call uses append = FALSE, so what does that mean? Ignore all of the other TRUE calls and treat them as FALSE (collapsing all of their arguments into one), or does it mean remove all appended `axis` calls and build off of the default (what it currently does)?